novu/api/integration.py

```diff
@@ -109,3 +109,18 @@ def limit(self, channel: Channel) -> IntegrationChannelUsageDto:
         return IntegrationChannelUsageDto.from_camel_case(
             self.handle_request("GET", f"{self._integration_url}/{channel}/limit")["data"]
         )
+
+    def set_primary(self, integration_id: str) -> bool:
+        """Set an integration as primary
+
+        Args:
+            integration_id: The integration ID
+
+        Returns:
+            Is integration set active
+        """
+        try:
+            res = self.handle_request("POST", f"{self._integration_url}/{integration_id}/set-primary")["data"]
+            return res.get("primary") is True
+        except requests.exceptions.HTTPError:
+            return False
```

Comment:
I still disagree with your proposal. I don't think it's actually a good idea to make the decision for the user how to handle errors.

If we want to warn our users that this could go wrong, then we might as well just specify it in the documentation. They will be able to define the procedure to follow in this case themselves.

We are a connector, not a decision-making tool.

```suggestion
    def set_primary(self, integration_id: str) -> IntegrationDto:
        """Set an integration as primary

        Args:
            integration_id: The integration ID

        Raises:
            HTTPError: Raise a 400 error when the integration identifier is poorly formulated
            HTTPError: Raise a 403 error when you are not authorized to do this action
            HTTPError: Raise a 404 error when provided integration identifier is not found in the application

        Returns:
            The instance of the updated integration
        """
        return IntegrationDto.from_camel_case(
            self.handle_request("POST", f"{self._integration_url}/{integration_id}/set-primary")["data"]
        )
```

Is you want to provide an utils

Reply:
I have made changes as per your suggestions. Also both testcases are modified accordingly.